docs/guidance/rayservice-troubleshooting.md

```diff
@@ -0,0 +1,81 @@
+# RayService troubleshooting
+
+RayService is a Custom Resource Definition (CRD) designed for Ray Serve. In KubeRay, it first creates a RayCluster and then proceeds to 
+create Ray Serve applications once the RayCluster is ready. If the issue pertains to the data plane, specifically your Ray Serve scripts 
+or Ray Serve configurations (`serveConfigV2`), troubleshooting may be challenging. This section provides some tips to help you debug these issues.
+
+## Observability
+
+### Method 1: Check KubeRay operator's logs for errors
+
+```bash
+kubectl logs $KUBERAY_OPERATOR_POD -n $YOUR_NAMESPACE | tee operator-log
+```
+
+The above command will redirect the operator's logs to a file called `operator-log`. You can then search for errors in the file.
+
+### Method 2: Check RayService CR status
+
+```bash
+kubectl describe rayservice $RAYSERVICE_NAME -n $YOUR_NAMESPACE
+```
+
+You can check the status and events of the RayService CR to see if there are any errors.
+
+### Method 3: Check logs of Ray Pods
+
+```bash
+kubectl exec -it $RAY_POD -n $YOUR_NAMESPACE -- bash
+# Check the logs under /tmp/ray/session_latest/logs
+```
+
+### Method 4: Check Dashboard
+
+```bash
+kubectl port-forward $RAY_POD -n $YOUR_NAMESPACE --address 0.0.0.0 8265:8265
+# Check $YOUR_IP:8265 in your browser
+```
+
+### Common issues
+
+#### Issue 1: Ray Serve script is incorrect.
+
+We strongly recommend that you test your Ray Serve script locally or in a RayCluster before
+deploying it to a RayService. [TODO: https://github.com/ray-project/kuberay/issues/1176]
+
+#### Issue 2: `serveConfigV2` is incorrect.
+
+For the sake of flexibility, we have set `serveConfigV2` as a string in the RayService CR.
+This implies that there is no strict type checking for the Ray Serve configurations in `serveConfigV2` field.
+Some tips to help you debug the `serveConfigV2` field:
+
+* Check [the documentation](https://docs.ray.io/en/latest/serve/api/#put-api-serve-applications) for the schema about
+the Ray Serve Multi-application API `PUT "/api/serve/applications/"`.
+* Unlike `serveConfig`, `serveConfigV2` adhere to the snake case naming convention. For example, `numReplicas` in `serveConfig` should be `num_replicas` in `serveConfigV2`. 
+
+#### Issue 3: The Ray image does not include the required dependencies.
+
+You have two options to resolve this issue:
```

Comment:
Would it be helpful here (and in the other issues where this may be relevant) to show an example of a sign/error that would help indicate to the user that this is the exact issue they are running into?

For instance, for the image not having the required dependencies, they will likely see `DEPLOY_FAILED` as the application status when using `kubectl describe rayservice`, and the status message should indicate that Serve failed to load the application because of some dependency issue.

Reply:
I added two examples for dependency issues.